### DISCUSSION TITLE
csv to shapefile source code for c#

### DISCUSSION THREAD

**Discussion by stevensrf11** [MaxRev-Dev/gdal.netcore#79] [Q&A] [answered]
I need to convert a csv file into a shapefile in a .Net6 windows application written in cshapr.
Does gdal have any source code available are assemblies I can used to perform such an operation

**Comment by MaxRev-Dev** [comment, marked as answer]
I don't think so - there is nothing in C# [samples](https://github.com/OSGeo/gdal/tree/master/swig/csharp/apps).
It will be easier to invoke ogr2ogr.
https://gis.stackexchange.com/a/276607
https://gdal.org/drivers/vector/csv.html

Another approach is to port some code from python:
https://gis.stackexchange.com/questions/202922/create-shapefile-from-csv-using-ogr-and-python
https://pcjericks.github.io/py-gdalogr-cookbook/vector_layers.html#read-a-csv-of-coordinates-as-an-ogrvrtlayer
Manually create a file and populate Shapefile:
https://pcjericks.github.io/py-gdalogr-cookbook/vector_layers.html#create-a-new-shapefile-and-add-data

**Reply by stevensrf11** [reply]
of thanks

**Comment by sindizzy** [comment]
You might try DotSpatial.

**Reply by MaxRev-Dev** [reply]
@sindizzy I have never tried DotSpatial, but I quickly went through the repo. I don't want to insult anyone, but it looks like something old and legacy without code samples and proper docs (CHM is dead, I found PDF for 1.7v only from issues). That library tries to bring spatial operations to Windows Forms, which are also superseded by the Web API or modern native app frameworks nowadays.
I'm using NTS and its ancestor JTS with Builder to test everything. It works for me well and has many features with Esri and GeoJson plugins.

**Reply by sindizzy** [reply]
It's updated to .NET Core 6 and has samples in the samples folder. Documentation was on codeplex and hasn't been redone yet. I translated it to wiki here https://github.com/sindizzy/DSW/wiki.
